test/functional/test_framework/test_framework.py

```diff
@@ -169,6 +169,7 @@ def parse_args(self):
         parser.add_argument("--cachedir", dest="cachedir", default=os.path.abspath(os.path.dirname(os.path.realpath(__file__)) + "/../../cache"),
                             help="Directory for caching pregenerated datadirs (default: %(default)s)")
         parser.add_argument("--tmpdir", dest="tmpdir", help="Root directory for datadirs")
+        parser.add_argument("--tmpdir-prefix", dest="tmpdir_prefix", help="Prefix to use for datadirs", default=TMPDIR_PREFIX)
```

Comment:
nit in the first commit: This makes the `--tmpdirprefix` option pretty useless (and confusing, given that there are now 3 tmpdir options). My recommendation would be to remove `--tmpdirprefix` when adding this option.

`--tmpdirprefix` is also silently ignored when `--tmpdir` is passed, so this could even make sense as a separate bug-fix PR

Reply:
I don't quite follow, `--tmpdirprefix` is for the test runner. If it can be left for a followup, I'd prefer to leave it for that.

Reply:
Any option for a single test will be passed up to the test runner. (Similar to how **kwargs can be used to pass through options).

If `--tmpdir` is used to set the root dir and `--tmpdir-prefix` to set the prefix folder name, and both options are passed up to test runner, then having a third and broken `--tmpdirprefix`, which only works for the test runner, seems increasingly confusing and there may even introduce new bugs.

Moreover, I wonder which later commit in this pull requires the first commit(s). Looking at 2a60a78a6483d59b3ea09377d663ca834abc7fcf, it looks like you are using the test runner to parallelize the test, which should already correctly set the unique seed id in the temp prefix folder name?

Reply:
I've added a commit which remove `--tmpdirprefix`

> Moreover, I wonder which later commit in this pull requires the first commit(s).

They're necessary for 88b63a6e163045e63a7540427834a57799234407 "tests: Automatically run both wallet types in parallel for wallet tests". The single invocation needs to have different tmpdirs for each wallet option it runs with. Although `--tmpdir-prefix` might not be needed with the last commit.

Reply:
Sorry, I think I gave the wrong commit. It should be a936a3f194f855e269a4dc2c25e17dbc55eb7a18 "tests: Run both descriptor and legacy wallet modes in single invocation".

The tmpdir changes are needed for this commit because the entire test is restarted for the other wallet type, so it needs to be able to make a new datadir.